File: Src/Icm.Core.Tests/Basic types extensions/BooleanExtensionsTest.vb
<TestFixture()>
Public Class BooleanExtensionsTest

    <TestCase(True, Result:=1)>
    <TestCase(False, Result:=0)>
    Public Function ToInteger_Test(bool As Boolean) As Integer
        Return bool.ToInteger
    End Function

    <TestCase(True, "cadena verdad", "cadena falso", "cadena nulo", Result:="cadena verdad")>
    <TestCase(False, "cadena verdad", "cadena falso", "cadena nulo", Result:="cadena falso")>
    <TestCase(Nothing, "cadena verdad", "cadena falso", "cadena nulo", Result:="cadena nulo")>
    Public Function IfN_Test(bool As Boolean?, trueString As String, falseString As String, nullString As String) As String
        Return bool.IfN(trueString, falseString, nullString)
    End Function

End Class

File: Src/Icm.Core.Tests/Basic types extensions/BooleanExtensionsTest.cs

using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using NUnit.Framework;
using Icm;

[TestFixture()]
public class BooleanExtensionsTest
{

	[TestCase(true, ExpectedResult = 1)]
	[TestCase(false, ExpectedResult = 0)]
	public int ToInteger_Test(bool @bool)
	{
		return @bool.ToInteger();
	}

	[TestCase(true, "cadena verdad", "cadena falso", "cadena nulo", ExpectedResult = "cadena verdad")]
	[TestCase(false, "cadena verdad", "cadena falso", "cadena nulo", ExpectedResult = "cadena falso")]
	[TestCase(null, "cadena verdad", "cadena falso", "cadena nulo", ExpectedResult = "cadena nulo")]
	public string IfN_Test(bool? @bool, string trueString, string falseString, string nullString)
	{
		return @bool.IfN(trueString, falseString, nullString);
	}

}
